Numerous Bug Fixes

nstearns96 opened the pull request:
Fixed Botany Soil incorrectly getting biome information for moisture and acidity (Fixes: #605 )
Fixed Compartment SearchDialog render bug  (Fixes: #610 )
Blooming Bee now recognizes all saplings as flowers (Fixes: #607 )
Blooming Bee bone meal effect now applies to all saplings
Fixed fence type name error (Fixes: #604 )
Fixed OsangeOrange typo (Fixes: #596 )
Fixed addFermentation typo
Fixed OsangeOrange sapling not having in-world texture (Fixes: #597 )
Fixed Avocado typo
Drinking from a stack of bottles no longer clears them all out, returning one empty bottle (Fixes: #603 )
Fixed Shift-clicking capsules in machines not consuming the capsule (Fixes: #564 )
Lit redstone ore now recognized as a flower for redstone bees
Lightning spawned from bees now plays sounds (Fixes: #570 )
 Flower mutations now work properly (Fixes: #405 )
Eliminated use of some deprecated functions
Shearing immature flowers no longer reduces durability of shears
Eliminated numerous texture warnings
Eliminated numerous prefix warnings
Fixed Rain Shield typo
Fixed Alveary typo
Fruit Press now gives correct amount of fluid output (Fixes: #612 )
Fluid Tiles in JEI now have proper names
Fixed Binnie Core bottle textures (Fixes: #262 )

nstearns96 committed:
Fixed Avocado typo
Fixed Botany Soil incorrectly getting biome information for moisture and acidity
Fixed Compartment SearchDialog render bug
Blooming Bee now recognizes all saplings
Blooming Bee bone meal effect now applies to all saplings
Fixed fence type name error
Fixed OsangeOrange typo
Fixed addFermentation typo
Fixed OsangeOrange sapling not having in-world texture

nstearns96 committed:
Drinking from a stack of bottles no longer clears them all out, returning one empty

nstearns96 commented:
Drinking from a stack of bottles no longer clears them all out, returning one empty bottle (Fixes: #603 )

nstearns96 committed:
Fixed Shift-clicking capsules in machines not consuming the capsule.
Lit redstone now recognized as a flower for redstone bees
Lightning spawned from bees now plays sounds

nstearns96 commented:
Fixed Shift-clicking capsules in machines not consuming the capsule (Fixes: #564 )
Lit redstone ore now recognized as a flower for redstone bees
Lightning spawned from bees now plays sounds (Fixes: #570 )

nstearns96 committed:
Flower mutations now work properly
Eliminated use of some deprecated functions
Shearing immature flowers no longer reduces durability of shears
Eliminated numerous texture warnings
Eliminated numerous prefix warnings
Fixed Rain Shield typo
Fixed Alveary typo
Fruit Press now gives correct amount of fluid output
Fluid Tiles in JEI now have proper names

nstearns96 commented:
Flower mutations now work properly (Fixes: #405 )
Eliminated use of some deprecated functions
Shearing immature flowers no longer reduces durability of shears
Eliminated numerous texture warnings
Eliminated numerous prefix warnings
Fixed Rain Shield typo
Fixed Alveary typo
Fruit Press now gives correct amount of fluid output (Fixes: #612 )
Fluid Tiles in JEI now have proper names
Fixed Binnie Core bottle textures (Fixes: #262 )

nstearns96 commented:
This is as much as I feel comfortable fixing for now. Anything more significant will require large rewrites or adding more content to the mod. I would appreciate it if I could get a review or approval for this request soon.

temp1011 commented on botany/src/main/java/binnie/botany/genetics/Flower.java:
```diff
@@ -253,10 +253,13 @@ public IFlower getOffspring(World world, BlockPos pos) {
+				float chance = mutation.getChance(world, pos, alleleFirst, alleleSecond, genome0, genome2);
+				if (chance > 0.0f && world.rand.nextFloat() * 100.0f < chance) {
+					template = BotanyCore.getFlowerRoot().templateAsChromosomes(mutation.getTemplate());
+					break;
```
Might be worth setting `parentFirst` here before breaking out of the loop for clarity?

nstearns96 replied:
Most likely, yes. I tried to maintain the structure of the original code as much as possible.

temp1011 replied:
fair enough

mezz replied:
the best way to make this code clear would be to separate this for-loop logic into a new function that returns a template or `null`

temp1011 commented on botany/src/main/java/binnie/botany/modules/ModuleFlowers.java:
```diff
@@ -112,7 +113,7 @@ public void registerItemsAndBlocks() {
 		Botany.proxy.registerItem(seed);
 
-		BinnieCore.getBinnieProxy().registerTileEntity(TileEntityFlower.class, "botany.tile.flower");
+		BinnieCore.getBinnieProxy().registerTileEntity(TileEntityFlower.class, new ResourceLocation("botany:tile.flower"));
```
Is this backwards compatible? Will TEs disappear from worlds if they are already placed due to the name change?

nstearns96 replied:
Looks like it's not backwards compatible, unfortunately

temp1011 replied:
Forestry had the same issue a while ago and we solved it with a datafixer: https://github.com/ForestryMC/ForestryMC/commit/9b4a4c2b1d0b495f180a4dfea2bcfe492546b4d0. Just have to be careful to test all the TEs are preserved

nstearns96 replied:
Great suggestion! Thanks for pointing that out to me.

nstearns96 replied:
The entities are backwards compatible now.

temp1011 commented on extrabees/src/main/java/binnie/extrabees/genetics/ExtraBeesFlowers.java:
```diff
@@ -121,7 +114,7 @@ public void register() {
 			}
 			case SAPLING: {
-				return Collections.singletonList(Blocks.SAPLING);
+				return ImmutableList.of(ModuleArboriculture.getBlocks().saplingGE, Blocks.SAPLING);
```
Perhaps use the ore dictionary here?

nstearns96 replied:
Good idea, works just fine!

temp1011 reviewed (commented):
Some potential world breakage here but otherwise things look good

temp1011 commented on extrabees/src/main/java/binnie/extrabees/worldgen/BinnieHiveDescription.java:
```diff
@@ -1,8 +1,19 @@
 package binnie.extrabees.worldgen;
 
+import java.util.ArrayList;
+import java.util.List;
```
These seem unnecessary?

nstearns96 replied:
Those were left over from trying to fix another bug, I'll get rid of them.

temp1011 commented:
You should probably move the 'fixes' lines from other comments to the first one so the issues are closed when this gets merged

temp1011 commented on extratrees/src/main/java/binnie/extratrees/machines/fruitpress/FruitPressLogic.java:
```diff
@@ -44,6 +44,9 @@ public ErrorState canProgress() {
 	protected void onFinishTask() {
+		final FluidStack tankFinish = FruitPressRecipeManager.getOutput(this.getUtil().getStack(FruitPressMachine.SLOT_CURRENT));
+		tankFinish.amount /= 25;
+		this.getUtil().fillTank((FruitPressMachine.TANK_OUTPUT), tankFinish);
```
After looking at the code a bit more I don't think this is the correct fix. The method `onUpdate` should output the correct amount over all the cycles but it looks like there are rounding/arithmetic errors. So I think the fix is in that method not this one.

Or maybe I'm missing something about how the fruit press works

nstearns96 replied:
You know, I was unsure of that fix. It was a bit hacky to be sure. I'll take another look at that and make it better.

nstearns96 replied:
That should be much better.

nstearns96 committed:
New Tile Entity names are now backward compatible
Fruit Press fix is properly implemented
Removed unnecessary imports
(Thanks to temp1011)

mezz commented:
Thanks for contributing! 
Please submit fixes in smaller batches next time. Small PRs are easier to review, and if some issue is blocking one PR it will not prevent all your other fixes from going through.

nstearns96 commented:
I'll be sure to. I was originally making these fixes for a private pack but decided to contribute to the mod directly instead. The initial commit was all of my fixes beforehand and it kinda snowballed from there.

mezz commented on extratrees/src/main/resources/assets/extratrees/lang/en_US.lang:
```diff
@@ -725,6 +725,112 @@ extratrees.fluid.miscfluid.sugarsyrup=Simple Syrup
 extratrees.fluid.miscfluid.agavenectar=Agave Nectar
 extratrees.fluid.miscfluid.grenadinesyrup=Grenadine Syrup
 
+## Fluid Tiles
```
I would like to avoid duplicating these localization keys, can you think of a better way?

nstearns96 replied:
That's how forestry does it. I don't think there's a way around it, it's due to how JEI shows fluid tiles.

mezz replied:
Hmm ok. We can fix this up some other time maybe, it's not a very big problem.

sealchan1 commented:
This is truly epic...I hope someone can now YouTube an up-to-date Botany mod spotlight and show all the basic flower breeding working as expected.
